recipes/openscenegraph/all/conanfile.py

```diff
@@ -102,163 +104,254 @@ def config_options(self):
+        tc.variables["OSG_WITH_ZEROCONF"] = False
+        tc.variables["OSG_WITH_LIBLAS"] = False
+        tc.variables["OSG_WITH_GIFLIB"] = self.options.get_safe("with_gif", False)
+        tc.variables["OSG_WITH_JPEG"] = self.options.get_safe("with_jpeg", False)
+        tc.variables["OSG_WITH_PNG"] = self.options.get_safe("with_png", False)
+        tc.variables["OSG_WITH_TIFF"] = self.options.with_tiff
+
+        if (self.options.get_safe("with_avfoundation")):
+            tc.variables["OSG_WITH_AV_FOUNDATION"] = True
 
         if self.settings.os == "Windows":
             # osg has optional quicktime support on Windows
-            cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_QuickTime"] = True
+            tc.variables["CMAKE_DISABLE_FIND_PACKAGE_QuickTime"] = True
+
+        tc.variables["OSG_MSVC_VERSIONED_DLL"] = False
 
-        cmake.definitions["OSG_MSVC_VERSIONED_DLL"] = False
+        if is_apple_os(self):
+            tc.preprocessor_definitions["GL_SILENCE_DEPRECATION"] = "1"
 
-        cmake.configure()
+        tc.generate()
 
-        return cmake
+        deps = CMakeDeps(self)
+        deps.set_property("freetype", "cmake_module_file_name", "Freetype")
+        deps.set_property("giflib", "cmake_file_name", "GIFLIB")
+        deps.generate()
+
+    def _patch_sources(self):
+        for package in ["Fontconfig", "Freetype", "GDAL", "GIFLIB", "GTA", "Jasper", "OpenEXR"]:
+            # Prefer conan's find package scripts over osg's
+            os.unlink(os.path.join(self.source_folder, "CMakeModules", f"Find{package}.cmake"))
+        plugins_root = Path(self.source_path.joinpath("src", "osgPlugins"))
+        for path in plugins_root.rglob("CMakeLists.txt"):
+            if path.parent == plugins_root:
+                # Don't replace in the root dir
+                continue
+            content = path.read_text()
+            # Correct usage of *_LIBRARY variables to *_LIBRARIES
+            content = content.replace("_LIBRARY", "_LIBRARIES")
+            # Allow explicit control of plugins via OSG_WITH_* variables
+            # e.g. replace IF(FFMPEG_FOUND) with IF(OSG_WITH_FFMPEG)
+            content = re.sub(r"\b([A-Z]+)_FOUND\b", r"OSG_WITH_\1", content)
+            path.write_text(content)
+        for path in self.source_path.joinpath(self.source_folder, "CMakeModules").rglob("*.cmake"):
+            content = path.read_text(encoding='utf-8', errors='ignore')
+            lib_match = re.search(r'FIND_LIBRARY\(([^ ]+)_LIBRARY', content)
+            if lib_match:
+                library_name = lib_match.group(1)
+                new_content = re.sub(rf'\b{library_name}_LIBRARY\b', rf'{library_name}_LIBRARIES', content)
+                path.write_text(new_content)
+
+        apply_conandata_patches(self)
+
+        # Not sure why, but CMake fails to find the EXPAT::EXPAT target created by Conan when Fontconfig is found as a module.
```

Comment:
Probably worth opening an issue for this.

Reply:
To OSG you mean? It looks like it's pretty much dead, last commit to master was over two years ago.

Reply:
No, I thought maybe it was an issue with the Expat Conan package. Maybe its not though.

Reply:
Ah, I see. Not really looked into it.
This patch and comment comes from the original PR that I based this PR on (https://github.com/conan-io/conan-center-index/pull/21355/files#diff-75b9f888a7159f587df53d73476654254af05ca90d777c2525335ec2d2c520d2R271-R273)

Reply:
Ran into this recently as well, it's because expat defines find mode to use `expat` and module mode to use `EXPAT`. If we had proper find_package aliases, we could use this unmodified. In my case I did a `replace_in_file`, `find_package(EXPAT)` -> `find_package(EXPAT MODULE)`. Not sure if a similar fix would work here, but maybe that info is helpful.